rfd/0034-desktop-access-windows.md

```diff
@@ -0,0 +1,268 @@
+eBPF on Linux) and enforce extra restrictions.
+
+```
++--------+
+| web UI |
++--------+
+    ^
+    | desktop protocol over websocket
+    v
++--------+
+| proxy  |
++--------+
+    ^
+    | desktop protocol over mTLS over reverse tunnel
+    |
++---|------------------------------+
+|   v                              |
+| +-------------------------+      |
+| | windows_backend_service |      |
+| +-------------------------+      |
+|   ^                              |
+|   | RDP over localhost           |
+|   v                              |
+| +-------------+                  |
+| | RDP service |                  |
+| +-------------+     Windows host |
++----------------------------------+
+```
+
+### Authentication
+
+In Windows, authentication is handled differently for standalone and
+Active Directory-enrolled machines. Standalone machines validate credentials
+using the local user database (SAM). Active Directory-enrolled machines send
+credentials to the domain controller for validation. Teleport Desktop Access
+will support both modes.
+
+Windows generally exposes 3 authentication mechanisms:
+- smart cards
+- username/password
+- kerberos tickets
+
+#### Smart cards
+
+Smart cards implement asymmetric cert-based authn, using PKI configured on the
+server/AD side. Since this is the only known way for us to support cert-based
+authn, Teleport will implement a smart card emulator to authenticate over RDP.
+This will be the primary authentication method.
+
+Smart card authentication and emulator are described in more detail in RFD 35.
+
+#### Username/password
+
+Username and password are most universal and Teleport should support these
+interactively as a fallback, when other methods are not available. This will
+allow users to replace their existing RDP client software, while providing RBAC
+and session recording, despite weaker authentication and worse UX.
+
+#### Kerberos tickets
+
+When a user is already authenticated against Kerberos (using one of the above
+methods), they get a Ticket-granting ticket (TGT) which can be used to get
+other tickets for specific services, such as an RDP host. Since Teleport has no
+exposure to Kerberos tickets (even when using AAD as SSO provider) and we can't
+assume that all clients use Windows machines, Teleport will not use TGTs for
+RDP authentication.
+
+### Host discovery
+
+When a user starts a new desktop session, they must specify the Windows host to
+connect to. Internally, Teleport tracks known Windows hosts using
+`WindowsDesktop` objects.
+
+There are 3 ways that `windows_desktop_service` discovers Windows hosts to
+register:
+- hardcoded list of standalone hosts provided in the config file (see
+  [configuration](#configuration))
+- list of Active Directory-enrolled hosts obtained from AD via LDAPS (LDAP over
+  SSL)
+  - LDAP library: https://pkg.go.dev/github.com/go-ldap/ldap/v3
+- local host, when running on a Windows machine in agent mode
```

Comment:
@r0mant is currently refactoring discovery for application access and DB access, to support dynamic registration of the apps and databases using API as well. This will enable both auto-discovery. Take a look [here](https://github.com/gravitational/teleport/pull/6787)

Reply:
Good idea. It should be easy to implement by exposing `WindowsDesktop` objects to `tctl create/rm`.
There are some questions about routing, but we can make simplifying assumptions like: if you dynamically register a windows machine, it has to be reachable from any `windows_desktop_service` running in gateway mode.